CHANGELOG.md

```diff
@@ -9,6 +9,8 @@
   - `RealmIOExcpetion` has been removed and replaced by `RealmFileException`.
 * Removed `RealmConfiguration.Builder(Context, File)` and `RealmConfiguration.Builder(File)` constructors.
 * `RealmConfiguration.Builder.assetFile(Context, String)` has been renamed to `RealmConfiguration.Builder.assetFile(String)`.
+* `Realm.createObject(Class<E>)` and `DynamicRealm.createObjec(String)` throws `RealmException` if they are called to create an object with a primary key defined.
```

Comment:
I would expand this a little I think. Something like

Objects with primary keys are now required to define it when the object is created. This means that `Realm.createObject(Class<E>)` and `DynamicRealm.createObject(String)` now throws `RealmException` if they are used to create an object with a primary key field. Use `createObject(Class, Object)` instead.